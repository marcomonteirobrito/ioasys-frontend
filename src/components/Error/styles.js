import styled from 'styled-components';

import Colors from '../../commons/Colors';

export const ErrorContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${Colors.backgroundErrorContainer};
  border-radius: 4px;
  width: 239px;
  height: 48px;
`;

export const ErrorMessageContainer = styled.div`
  color: ${Colors.white};
  font-weight: 700;
`;

export const ErrorArrow = styled.div`
  background: ${Colors.backgroundErrorContainer};
  width: 16px;
  height: 16px;
  margin: 16px 0px -8px 16px;
  transform: rotate(45deg);
`;
