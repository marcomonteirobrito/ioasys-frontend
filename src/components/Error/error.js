import React from 'react';
import Proptypes from 'prop-types';

import { ErrorMessageContainer, ErrorContainer, ErrorArrow } from './styles';

const ErrorMessage = ({ id, className, message }) => (
  <>
    <ErrorArrow />
    <ErrorContainer id={id} className={className}>
      <ErrorMessageContainer>{message}</ErrorMessageContainer>
    </ErrorContainer>
  </>
);

ErrorMessage.propTypes = {
  message: Proptypes.string.isRequired,
  className: Proptypes.string,
  id: Proptypes.string,
};

ErrorMessage.defaultProps = {
  className: 'error-message',
  id: 'error-message',
};

export default ErrorMessage;
