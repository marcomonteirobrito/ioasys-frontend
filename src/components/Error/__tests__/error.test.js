/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';

import Error from '../Error';

const renderError = () =>
  render(
    <Error id="error" className="error-container" message="Erro no login" />
  );

describe('Error', () => {
  it('Deve renderizar o componente corretamente', () => {
    renderError();

    expect(document.body).toMatchSnapshot();
  });

  it('Deve renderizar mensagem corretamente', () => {
    renderError();

    const errorMessage = screen.queryByPlaceholderText('Erro no login');

    expect(errorMessage).toBeDefined();
  });
});
