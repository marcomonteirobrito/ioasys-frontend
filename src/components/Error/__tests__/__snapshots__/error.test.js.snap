// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Deve renderizar o componente corretamente 1`] = `
<body>
  <div>
    <svg
      class="sc-dlfnbm hFSAlI"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 10.44l-.413.56H2.393L2 10.46 7.627 5h.827L14 10.44z"
      />
    </svg>
    <div
      class="sc-bdfBwQ jrIbtW error-container"
      id="error"
    >
      <div
        class="sc-gsTCUz bEBqSC"
      >
        Erro no login
      </div>
    </div>
  </div>
</body>
`;
